### DISCUSSION TITLE
algorithms/ads/15

### DISCUSSION THREAD

**Discussion by giscus** [NoughtQ/notebook#23] [Announcements]
# algorithms/ads/15

NoughtQ的笔记本

https://note.noughtq.top/algorithms/ads/15

**Comment by TillyEndless** [comment]
Replacement Selection的例一step3，似乎判断的是12<81?(上面写的是12<94)
因为前面说是比较即将删除和即将放入的元素大小。此外，如果是删除后下一步删除的元素和上一步即将放入的元素比较，即94>12，则run2不应该包含28

**Reply by NoughtQ** [reply]
- step 3 那步确实是我搞错了，应该比较的是12和81
- 按照“比较即将删除和即将放入的元素大小”这一策略，28应该与17做比较（我笔记里的gif可能放的太快了，建议看PPT），因为17从堆出去后，28就进来了，而28比17大，因此28还是会放在run 2里的

**Reply by TillyEndless** [reply]
是这样滴owo
